_posts/2018-07-27-Our-first-look-at-debos-new-Debian-images-generator.md

```diff
@@ -0,0 +1,285 @@
+2018/07/26 13:54:37 ==== debootstrap ====
+2018/07/26 13:54:37 Debootstrap | E: debootstrap can only run as root
+2018/07/26 13:54:37 debootstrap.log | cat: /home/osboxes/.debos-418422200/root/debootstrap/debootstrap.log: No such file or directory
+2018/07/26 13:54:37 Action `debootstrap` failed at stage Run, error: exit status 1
+```
+
+Of course. I don't mind running as `root` to see how far can we go this time (we
+are in `VirtualBox` after all):
+
+```
+sudo ~/go/bin/debos ~/go/src/github.com/go-debos/debos/doc/examples/example.yaml
+```
+
+The privileged run failed at:
+
+```
+2018/07/26 13:36:38 Debootstrap (stage 2) | chroot: failed to run command ‘/debootstrap/debootstrap’: Exec format error
+2018/07/26 13:36:38 debootstrap.log | gpgv: Signature made Thu Jul 26 10:21:51 2018 EDT
+2018/07/26 13:36:38 debootstrap.log | gpgv:                using RSA key A1BD8E9D78F7FE5C3E65D8AF8B48AD6246925553
+2018/07/26 13:36:38 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (7.0/wheezy) <ftpmaster@debian.org>"
+2018/07/26 13:36:38 debootstrap.log | gpgv: Signature made Thu Jul 26 10:21:51 2018 EDT
+2018/07/26 13:36:38 debootstrap.log | gpgv:                using RSA key 126C0D24BD8A2942CC7DF8AC7638D0442B90D010
+2018/07/26 13:36:38 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (8/jessie) <ftpmaster@debian.org>"
+2018/07/26 13:36:38 debootstrap.log | gpgv: Signature made Thu Jul 26 10:21:51 2018 EDT
+2018/07/26 13:36:38 debootstrap.log | gpgv:                using RSA key 16E90B3FDF65EDE3AA7F323C04EE7237B7D453EC
+2018/07/26 13:36:38 debootstrap.log | gpgv: Good signature from "Debian Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>"
+2018/07/26 13:36:38 Action `debootstrap` failed at stage Run, error: exit status 126
+```
+
+It seems to me that no correct `qemu-static` is called in order to run
+`arm64` binary.
+
+When building for the host archtiecture:
+
+```
+sudo ~/go/bin/debos  -t architecture:"amd64" ~/go/src/github.com/go-debos/debos/doc/examples/example.yaml
+```
+
+After a while, **almost** success:
+
+```
+2018/07/26 14:00:39 Debootstrap | I: Base system installed successfully.
+2018/07/26 14:00:39 Action `debootstrap` failed at stage Run, error: exec: "systemd-nspawn": executable file not found in $PATH
+```
+
+Another package missing, namely `systemd-container`:
+
+```
+sudo apt install systemd-container
+```
+
+And finally, we have our image:
+
+```
+2018/07/26 14:11:45 ==== overlay ====
+Overlaying /home/osboxes/go/src/github.com/go-debos/debos/doc/examples/overlays/sudo on /home/osboxes/.debos-414397177/root
+2018/07/26 14:11:45 ==== run ====
+2018/07/26 14:11:45 ==== pack ====
+2018/07/26 14:11:45 Compression to /home/osboxes/debian-stretch-amd64.tgz
+2018/07/26 14:11:56 ==== Recipe done
+```
+
+```
+88M     debian-stretch-amd64.tgz
+```
+
+## Third approach - docker container
+
+With above problems in mind, it seems like mandatory to package all those
```

Comment:
`With the above problems in mind...` – since the problems are known to the user (are mentioned above).
In general, I would suggest using one form – either official i.e. `I am`, or unofficial i.e. `I'm `.